### DISCUSSION TITLE
如何开启CSP？

### DISCUSSION THREAD

**Discussion by gaoyia** [tusen-ai/naive-ui#3191] [Q&A]
如何开启CSP？

**Comment by gaoyia** [comment]
css都是直接用的内部样式，开启CSP无法生效

**Reply by 07akioni** [reply]
我不太熟这个东西，现在是遇到了什么问题，有啥资料可以参考吗？

**Reply by gaoyia** [reply]
<https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP>

```html
<meta http-equiv="Content-Security-Policy" content="default-src 'self'; script-src 'self'; style-src 'self'">
```

**Reply by 07akioni** [reply]
开启无法生效是怎么说？

**Reply by 07akioni** [reply]
我理解 naive 没有 css 文件，全都是 js，因此也没有必要设定 style-src

**Reply by 07akioni** [reply]
只需要设定 script-src

**Reply by gaoyia** [reply]
这个逻辑是错的，并不是项目没用上css文件就不需要style-src。
CSP是用来预防xss攻击插入的代码引发的安全问题。插入的代码是不可控的。
不过插入style代码造成的安全问题相对来说可能更容易发现，危害也更小。
这个问题可能是受到css-in-js技术的限制吧，估计无解。

**Reply by 07akioni** [reply]
我再研究研究

**Comment by tu6ge** [comment]
这个 style-src 应该怎么设置才能正常呢

我最近在使用 naive-ui 的项目，准备使用 tauri 打包成 android 在设置 CSP 的时候遇到了样式错乱的问题 @07akioni
